packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/config_loader.py

```
@@ -141,35 +141,26 @@ def build_config_from_dict(config_data: dict[str, Any]) -> Config:
 
     # Deprecation: detect legacy blocks and log a warning once
     legacy_embedding = global_data.get("embedding")
+
+    if legacy_embedding:
+        raise ValueError(
+            "Configuration error: 'global.embedding' is no longer supported. "
+            "Please migrate your configuration to 'global.llm'."
+        )
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)